### PR TITLE
docs(contribution-guidelines): add best practices section for coding guidelines

### PR DESCRIPTION
**Pull request by AshwinSri23**
# 🚩 Linked issue

The issue is 104. The issue can be found [here.](https://github.com/cuhacking/2025/issues/104).

## 🛠 How does this PR address the issue(s)? How does it solve the problem?

The following PR is for adding a best practices section to the coding guidelines page. Best practices would describe to new contributors conventions for different areas within cuHacking's development infrastructure. 
These areas are: 
- Testing 
- Frontend
- Backend
- API Design
- Docs
- Features

# ✅ Checklist

Make sure your PR follows the according checklist:

### **Code Quality**

<table>
  <thead>
    <tr>
      <th>Done</th>
      <th>Checklist Item</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><input type="checkbox" /></td>
      <td>Code is clean and readable (functions used appropriately)</td>
    </tr>
    <tr>
      <td><input type="checkbox" /></td>
      <td>Code is commented appropriately (no unnecessary comments, follows guidelines)</td>
    </tr>
    <tr>
      <td><input type="checkbox" /></td>
      <td>Code is DRY (no repeated code)</td>
    </tr>
    <tr>
      <td><input type="checkbox" /></td>
      <td>Functions are named appropriately (camelCase)</td>
    </tr>
    <tr>
      <td><input type="checkbox" /></td>
      <td>Variables are named appropriately (camelCase)</td>
    </tr>
    <tr>
      <td><input type="checkbox" /></td>
      <td>Files are named appropriately (kebab-case)</td>
    </tr>
    <tr>
      <td><input type="checkbox" /></td>
      <td>Code performs as expected (no errors or issues when running)</td>
    </tr>
    <tr>
      <td><input type="checkbox" /></td>
      <td>Code is committed and branched accordingly</td>
    </tr>
  </tbody>
</table>

### **PR Review Readiness**

<table>
  <thead>
    <tr>
      <th>Done</th>
      <th>Checklist Item</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><input type="checkbox" /></td>
      <td>PR is linked to an issue (if applicable)</td>
    </tr>
    <tr>
      <td><input type="checkbox" /></td>
      <td>PR is clear and comprehensible</td>
    </tr>
    <tr>
      <td><input type="checkbox" /></td>
      <td>PR is ready for review (if not, consider using a draft PR)</td>
    </tr>
    <tr>
      <td><input type="checkbox" /></td>
      <td>PR title follows template guidelines</td>
    </tr>
    <tr>
      <td><input type="checkbox" /></td>
      <td>PR description follows template guidelines</td>
    </tr>
  </tbody>
</table>

## Summary by CodeRabbit

- **New Features**
	- Introduced a comprehensive section on best practices for end-to-end (E2E) testing in the coding guidelines.
	- Detailed organization of E2E tests for different devices (desktop, mobile, tablet).
	- Provided instructions for running tests and managing configurations.
	- Expanded guidelines on backend practices, including commit descriptions and issue resolution.

- **Documentation**
	- Added a new "Best Practices" section with subsections on testing, backend, features, and API best practices.

### CODE REVIEW

File: apps/docs/src/content/docs/contribution-guidelines/coding-guidelines.mdx
Comment (HasithDeAlwis) [inline apps/docs/src/content/docs/contribution-guidelines/coding-guidelines.mdx:1]: Content is ok, but these should be their own directories/pages. Can you refer to how Farabi did his, [here](https://github.com/cuhacking/2025/pull/144)

File: apps/docs/src/content/docs/contribution-guidelines/coding-guidelines.mdx [hunk 1]
```diff
@@ -83,3 +83,91 @@ Now take this concept and apply it to code. You'll find that the parallels are u
...
+
+### **Design**
+
+
+
+
+
+
+
+
```
[the remark this answers is not in the document]
Comment (HasithDeAlwis) [reply apps/docs/src/content/docs/contribution-guidelines/coding-guidelines.mdx:173]: 👀

**Review by HasithDeAlwis — CHANGES_REQUESTED** [review]
Nice work on your first PR! Just gotta get it to the finish line now and make a couple changes, good work 👍

**Review by JeremyFriesenGitHub — COMMENTED** [review]
see discord private message abt design

**Comment by MFarabi619** [comment]
Closing as stale

**Pull request merged**